### ISSUE TITLE
Completion behavior if the word is already in the editor

### ISSUE THREAD

**Issue by KirillOsenkov** [mhutch/MonoDevelop.MSBuildEditor#227]
If the caret is not at the end of the word, it completes the remainder of the word (unlike C#):
![image](https://github.com/mhutch/MonoDevelop.MSBuildEditor/assets/679326/ecaa8323-a1a8-4723-a02d-7b4f3a43181b)

Not sure if this is the desired behavior.

Maybe if there's a single completion, and the word in the editor around the trigger point matches the completion exactly, the completion shouldn't show but the caret should move to the end of the word? This is how C# works.

**Comment by mhutch** [comment]
Yes, the trigger logic for MSBuild expressions doesn't yet readahead to calculate the full extent of the trigger location.
